foundations/foundation-vertx/src/main/java/org/apache/servicecomb/foundation/vertx/metrics/DefaultHttpClientMetrics.java

```diff
@@ -133,7 +144,18 @@ public void disconnected(Object webSocketMetric) {
 
   @Override
   public DefaultHttpSocketMetric connected(SocketAddress remoteAddress, String remoteName) {
-    return new DefaultHttpSocketMetric(null);
+
+    DefaultHttpSocketMetric socketMetric = new DefaultHttpSocketMetric(null);
+    try {
+      DefaultHttpClientMetrics clientMetrics = (DefaultHttpClientMetrics) ((HttpClientImpl) client).getMetrics();
+      DefaultClientEndpointMetric clientEndpointMetric = clientMetrics.clientEndpointMetricManager
+          .getClientEndpointMetricMap().get(remoteAddress);
+      // set endPointMetric when use http2
+      socketMetric.setEndpointMetric(clientEndpointMetric);
```

Comment:
just clean all logic in endpointConnected, move them to this place
and add comments to describe why not just write them in endpointConnected
and not care if vertx fix this bug, just use the new logic is no problem.

Reply:
ok, I will finish it soon